modules/backup/README.md

```diff
@@ -0,0 +1,74 @@
+# GCP CloudSQL Backup
+
+## Internal Backups
+
+Cloud SQL provides an [internal default backup options](https://cloud.google.com/sql/docs/mysql/backup-recovery/backups#automated-backups).
+This supports standard use-cases. (One Backup ~ every 24h stored for up to 365 days).
+
+For many use cases this might be enough but for many other not:
+e.g. for SQL Server there is currently no Point-in-time Recovery (PITR) available so you might want to take more updates.
+Or your instance is not that critical and to [save costs](https://cloud.google.com/sql/docs/mysql/backup-recovery/backups#what_backups_cost) you are fine with one backup every 7 days.
+
+> **Note**: To enable or disable the internal backups use the Terraform variable `enable_internal_backup`
+
+For internal backups the workflow also takes care about deleting the old backups.
+
+## Exports to GCS
+
+But the backups mentioned above are bound to the SQL instance.
+This means as soon as you [delete the instance](https://cloud.google.com/sql/docs/mysql/delete-instance) you are also loosing your backups.
+To avoid this you can also export data. There is a second workflow doing exactly that.
+
+> **Note**: To enable or disable the exports to GCS Buckets use the Terraform variable `enable_export_backup`
+
+### Deleting old exports
+
+The export workflow does not take care about deleting old backups. Please take care about that yourself.
+You can use [Object Lifecycle Management](https://cloud.google.com/storage/docs/lifecycle) to delete old exports.
+
+## Required APIs
+
+- `workflows.googleapis.com`
+- `cloudscheduler.googleapis.com`
+
+## Monitoring
+
+Monitoring your exports/backups is not part of this module.
+Please ensure that you monitor for failed workflows to ensure you have regular backups/exports.
+
+A good metric could be to monitor for workflow executions that not succeeded:
+
+```mql
+fetch workflows.googleapis.com/Workflow
+| metric 'workflows.googleapis.com/finished_execution_count'
+| filter (metric.status != 'SUCCEEDED')
+| group_by 1d,
+    [value_finished_execution_count_mean: mean(value.finished_execution_count)]
+| every 1d
+| condition val() > 0 '1'
+```
+
+## Inputs
```

Comment:
We should add these markers https://github.com/terraform-google-modules/terraform-google-sql-db/blob/fe91aa5093c03b8cc823c2f1f43c18f0b9d0812a/modules/mssql/README.md?plain=1#L7 which lets automation generate these docs via `make docker_generate_docs`